pkg/diagnostics/report.go

```diff
@@ -12,26 +13,40 @@ import (
 
 // RunReport runs a CLI command
 func RunReport(cmd *cobra.Command, args []string) {
+	log := util.Logger()
+
+	// Fetching coreApp configurations
+	coreApp := util.KubeObject(bundle.File_deploy_internal_statefulset_core_yaml).(*appsv1.StatefulSet)
+	coreApp.Namespace = options.Namespace
+	if !util.KubeCheck(coreApp) {
+		log.Fatalf(`❌ Could not get core StatefulSet %q in Namespace %q`,
+			coreApp.Name, coreApp.Namespace)
+	}
+
+	// Fetching endpoint configurations
+	endpointApp := util.KubeObject(bundle.File_deploy_internal_deployment_endpoint_yaml).(*appsv1.Deployment)
+	endpointApp.Namespace = options.Namespace
+	if !util.KubeCheck(endpointApp) {
+		log.Fatalf(`❌ Could not get endpoint Deployment %q in Namespace %q`,
+			endpointApp.Name, endpointApp.Namespace)
+	}
```

Comment:
This duplication can be a function that gets a kubernetes object and is called twice, once for core sts and a second time for endpoint deployment.
@achouhan09 Create a new PR that consolidates those into a single function.